python/pyspark/shell.py

```diff
@@ -44,9 +44,9 @@
         .enableHiveSupport()\
         .getOrCreate()
 except py4j.protocol.Py4JError:
-    spark = SparkSession(sc)
+    spark = SparkSession(SparkContext.getOrCreate())
```

Comment:
actually, we should use the builder pattern here instead: `SparkSession.builder.getOrCreate()`